src/ToggleSwitch.tsx

```diff
@@ -0,0 +1,333 @@
+import React, {MouseEventHandler, useCallback, useEffect} from 'react'
+import styled, {css} from 'styled-components'
+import {variant} from 'styled-system'
+import {Box, Spinner, Text} from '.'
+import {get} from './constants'
+import {useProvidedStateOrCreate} from './hooks'
+import sx, {BetterSystemStyleObject, SxProp} from './sx'
+import VisuallyHidden from './_VisuallyHidden'
+import {useTheme} from './ThemeProvider'
+
+const TRANSITION_DURATION = '80ms'
```

Comment:
It may just be me, but this felt surprisingly fast. Is this aligned to our other motion-related timings across Primer? Wondering if we have a design token for it somewhere too 🤔 

If it's staying at 80ms, can we include a slower version for users that `prefers-reduced-motion`? https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion

Reply:
80ms is being proposed for the "fast" duration in our new tokens. See [Transition section of the doc](https://docs.google.com/document/d/1Yc5WKc0R0mAQFDPOsS9HgLM7ZFB0lNJHWeeycbi5mvY/edit#heading=h.2q6g3fn7042u) (only available to hubbers).

Good call on `prefers-reduced-motion`. We should turn off the transition entirely for those users.